test/test.js

```diff
@@ -563,12 +595,76 @@ describe('finalhandler(req, res)', function () {
         error = e
       }
 
-      request(createServer(err, { onerror: log }))
-        .get('/')
+      wrapper(request(createServer(err, { onerror: log }))
+        .get('/'))
         .end(function () {
           assert.equal(error, err)
           done()
         })
     })
   })
-})
+
+  describe('no deprecation warnings', function () {
+    it('should respond 404 on no error', function (done) {
+      var warned = false
+
+      process.once('warning', function (warning) {
+        if (/The http2 module is an experimental API/.test(warning)) return
+        assert.fail(warning)
+      })
+
+      wrapper(request(createServer())
+        .head('/foo'))
+        .expect(404)
+        .end(function (err) {
+          assert.strictEqual(warned, false)
+          done(err)
+        })
+    })
+
+    it('should respond 500 on error', function (done) {
+      var warned = false
+
+      process.once('warning', function (warning) {
+        if (/The http2 module is an experimental API/.test(warning)) return
+        assert.fail(warning)
+      })
+
+      var err = createError()
+
+      wrapper(request(createServer(function (req, res) {
+        var done = finalhandler(req, res)
+
+        if (typeof err === 'function') {
+          err(req, res, done)
+          return
+        }
+
+        done(err)
+      }))
+        .head('/foo'))
+        .expect(500)
+        .end(function (err, res) {
+          assert.strictEqual(warned, false)
+          done(err)
+        })
+    })
+  })
+}
+
+var servers = [
+  ['http', createHTTPServer]
+]
+
+var nodeVersion = process.versions.node.split('.').map(Number)
+
+// `superagent` only supports `http2` since Node.js@10
+if (http2 && nodeVersion[0] >= 10) {
```

Comment:
`superagent` (under the hood of `supertest`) supports `http2` since Node.js@10, for old versions you will get:
```
Error: superagent: this version of Node.js does not support http2
```